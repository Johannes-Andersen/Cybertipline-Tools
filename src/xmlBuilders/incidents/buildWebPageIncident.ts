import { XMLBuilder } from 'fast-xml-parser';
import type { WebPageIncident } from '../../types';

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildWebPageIncident = (
  webPageIncident: WebPageIncident,
  keyName = 'webPageIncident',
): string =>
  builder.build({
    [keyName]: {
      url: webPageIncident.url,
      additionalInfo: webPageIncident.additionalInfo,
      '@_thirdPartyHostedContent': webPageIncident.thirdPartyHostedContent,
    },
  });
