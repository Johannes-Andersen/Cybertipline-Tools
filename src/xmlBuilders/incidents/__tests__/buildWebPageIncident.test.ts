import { describe, expect, it } from 'vitest';
import type { WebPageIncident } from '../../../types';
import { buildWebPageIncident } from '../buildWebPageIncident';

describe('buildWebPageIncident', () => {
  it('should build a web page incident', () => {
    const webPageIncident: WebPageIncident = {
      additionalInfo: 'This is a test',
      thirdPartyHostedContent: true,
      url: ['https://example.com', 'https://example.org'],
    };

    const result = buildWebPageIncident(webPageIncident);

    expect(result).toMatchSnapshot();
  });
});
