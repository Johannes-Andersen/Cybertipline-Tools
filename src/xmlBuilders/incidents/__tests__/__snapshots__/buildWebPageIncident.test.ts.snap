// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildWebPageIncident > should build a web page incident 1`] = `"<webPageIncident thirdPartyHostedContent="true"><url>https://example.com</url><url>https://example.org</url><additionalInfo>This is a test</additionalInfo></webPageIncident>"`;
